vllm/v1/engine/async_llm.py

```diff
@@ -439,7 +440,17 @@ async def output_handler():
                     outputs = await engine_core.get_output_async()
                     num_outputs = len(outputs.outputs)
 
-                    iteration_stats = IterationStats() if (
+                    if outputs.mfu_output:
+                        mfu_info = MFUInfo(
+                            flops=outputs.mfu_output.flops,
+                            read_bytes=outputs.mfu_output.read_bytes,
+                            write_bytes=outputs.mfu_output.write_bytes,
+                            latency_s=outputs.mfu_output.latency_sec,
+                        )
+                    else:
+                        mfu_info = None
+
+                    iteration_stats = IterationStats(mfu_info=mfu_info) if (
```

Comment:
Yeah, the purpose of `IterationStats` is to aggregate data as `process_outputs()` iterates across the per-request `EngineCoreOutput` list

None of that aggregation is needed here - if the MFU stats was in `SchedulerStats` you can just use it from there in the logger's `record()` method